### DISCUSSION TITLE
Where is original text data stored?

### DISCUSSION THREAD

**Discussion by etjm1993** [FlowiseAI/Flowise#859] [Q&A]
Hi there!

When I upload a document to Flowise and then upsert it to Pinecone, where is the original text stored and then how is it accessed when I ask a question to the chat?

I want to create a process of automatically creating embeddings and upserting them to pinecone so that I do not have to do it manually but I am struggling to find the answer to my question above. 

Any help or direction would be highly appreciated!

Thank you

**Comment by HenryHengZJ** [comment]
The original text is converted into a base64 encoded string and stored as part of your chatflow data. When you ask a question in the chat when using any upsert node, it will automatically create embeddings and upsert data

**Reply by jeffrey-devhkg** [reply]
Is this the same with pdf file upload?

**Reply by chungyau97** [reply]
Hi @jeffrey-devhkg, yes it's the base64 of the pdf file upload.

**Reply by jeffrey-devhkg** [reply]
Thanks!

**Reply by jeffrey-devhkg** [reply]
Is the encoded base64 of the pdf stored in database.sqlite?

**Reply by HenryHengZJ** [reply]
yes

**Comment by etjm1993** [comment]
Got it, I am looking for a way to automate the process of creating
embeddings and upserting the data whilst retaining the original text. Is
there a way to do this with flowise?

Thanks!

**Reply by HenryHengZJ** [reply]
This is not possible yet on Flowise, definitely hope to build this feature in future

**Comment by etjm1993** [comment]
No it was with json documents.

The way I solved it was to index them into vectara and then they are ready
to use via flowise without any embeddings or storing of the text elsewhere!

**Reply by chungyau97** [reply]
Hi @etjm1993, Vectara itself have built-in embedding on their platform.
